src/components/Main.js

```diff
@@ -35,69 +50,64 @@ export default function Main() {
     };
   }, []);
 
-  const coinName = Object.keys(coinDataList);
-  const coinNameList = coinName.slice(0, coinName.length - 1);
+  const filteredCoinList =
+    searchCoin === ""
+      ? coinList
+      : coinList.filter((coin) => coin.currency_name === searchCoin);
 
-  const coinCurrentPrice = Object.values(coinDataList).map((coinPrice) => {
-    return coinPrice["closing_price"];
-  });
-  const coinCurrentPriceList = coinCurrentPrice.slice(
-    0,
-    coinCurrentPrice.length - 1
-  );
+  const handleClickSearch = () => {
+    const coinName = document.getElementById("coin-search").value;
+    setSearchCoin(coinName);
+  };
 
-  const coinChangeRate = Object.values(coinDataList).map((changeRate) => {
-    return changeRate["fluctate_rate_24H"];
-  });
-  const coinChangeRateList = coinChangeRate.slice(0, coinChangeRate.length - 1);
+  const handleKeyUpSearch = (e) => {
+    if (e.key === "Enter") {
+      const coinName = e.target.value;
+      setSearchCoin(coinName);
+    }
+  };
```

Comment:
```suggestion
  const handleKeyUpSearch = (e) => {
    const coinName = e.target.value;
    
    if (e.key === "Enter") {
      setSearchCoin(coinName);
    }
  };
```
제 개인적인 의견이지만 coinName을 선언하는 구문을 if 문 밖으로 빼주면 좀 더 가독성이 좋아질 거 같다는 생각이 듭니다. !

e.key === "Enter" 일 때 변수 선언을 하고 setSearchCoin(coinName) 둘 다 해주는 거보다
미리 coinName을 선언해놓고 e.key가 "Enter"일 때 setSearchCoin(coinName)만 하는 게 조금 더 자연스럽단 생각이 들어서 코멘트 남깁니다. !!